src/js/components/TimelineUI.js

```diff
@@ -0,0 +1,51 @@
+import React, { Component } from "react";
+import TweetBlock from "./TweetBlock";
+import {fetchHomeTimeline} from "../twitter-api.js";
+import "../../css/components/TimelineUI.scss";
+
+// Presentational Component for Timeline
+export default class TimelineUI extends React.Component {
+
+	constructor() {
+		super();
+		this.state = {
+			tweets: null,
+			error: null
+		}
+	}
+
+	componentDidMount() {
+		this.updateTimeline();
+	}
+
+	render() {
+		let displayedElem; // Either display error or tweets
+		if (this.state.error) {
+			displayedElem = <div id="error-div">{this.state.error}</div>
+		}
+		else if (this.state.tweets) {
+			displayedElem = 
+				<div id="tweets">{
+					this.state.tweets.map((tweet, index) => 
```

Comment:
do you need index here?

this is kind of hard to read, are you declaring another variable here?

Reply:
I'm creating a React component for the tweets and assigning it to displayedElem here. I agree it's hard to read. I changed this to the code snippet below; is this easier to read?
`
const tweets = this.state.tweets.map(tweet =>
	<div key={tweet.url} className="row">
		<TweetBlock tweet={tweet}/>
	</div>
);
displayedElem = <div id="tweets">{tweets}</div>;
`

Reply:
I'm creating a React component for the tweets and assigning it to displayedElem here. I agree it's hard to read. Does the updated code look cleaner now?

Reply:
👍